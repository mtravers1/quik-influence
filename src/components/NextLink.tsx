import { Link as ChakraLink, LinkProps } from "@chakra-ui/layout";
import Link from "next/link";
import React from "react";

export type NextLinkProps = Omit<LinkProps, "as"> & {
  href: string;
  as?: string;
  isInternal?: boolean;
  onClick?: () => void;
  hardRefresh?: boolean;
};

const NextLink: React.FC<NextLinkProps> = ({
  href,
  isInternal = true,
  onClick = () => {},
  as,
  children,
  hardRefresh = false,
  ...rest
}) => {
  const hoverStyle = {
    color: "red.500"
  };

  if (isInternal) {
    return hardRefresh ? (
      <ChakraLink href={href} onClick={onClick} _hover={hoverStyle} {...rest}>
        {children}
      </ChakraLink>
    ) : (
      <Link href={href} as={as} passHref>
        <ChakraLink onClick={onClick} _hover={hoverStyle} {...rest}>
          {children}
        </ChakraLink>
      </Link>
    );
  }

  let finalHref: string;
  const pathToUrl = as ?? href;
  if (/^(https|http|mailto)/.test(pathToUrl)) {
    finalHref = pathToUrl;
  } else {
    finalHref = `/${pathToUrl}`;
  }

  return (
    <ChakraLink
      href={finalHref}
      _hover={hoverStyle}
      onClick={onClick}
      {...rest}
    >
      {children}
    </ChakraLink>
  );
};

export default NextLink;
