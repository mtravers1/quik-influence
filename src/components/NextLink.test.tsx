import { render } from '@testing-library/react'
import NextLink from './NextLink';

it('renders NextLink component', () => {
  const { container } = render(<NextLink href={'/#'} />)
  expect(container).toMatchSnapshot()
});
