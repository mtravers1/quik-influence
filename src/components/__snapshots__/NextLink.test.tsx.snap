// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NextLink component 1`] = `
<div>
  <a
    class="chakra-link css-1trwvm"
    href="/#"
  />
</div>
`;

exports[`renders NextLink component 2`] = `
<div>
  <a
    class="chakra-link css-1trwvm"
    href="https://www.google.com"
  />
</div>
`;

exports[`renders NextLink component 3`] = `
<div>
  <a
    class="chakra-link css-1trwvm"
    href="/dashboard"
  />
</div>
`;

exports[`renders NextLink component 4`] = `
<div>
  <a
    class="chakra-link css-1trwvm"
    href="/dashboard"
  />
</div>
`;
